### ISSUE TITLE
[Feature Request] Pause new git action if fork is already running a git action

### ISSUE THREAD

**Issue by darkdescendant** [ForkIssues/Tracker#1976]
I work in a large code base, and often fork is running some git action that takes a while that I didn't initiate. If I try to do something in that repo that requires git, fork will throw an error because it can't get necessary exclusive access. This is almost always because fork is fighting itself.

It would be nice to have fork wait for any current git actions for the repo in question that it initiated and is waiting on and then run the action requested rather than just letting it fail out of hand.
